source/includes/steps-fle-create-data-encryption-key.yaml

```diff
@@ -11,19 +11,17 @@ content: |
         :tabid: java-sync
 
         .. code-block:: java
-           :emphasize-lines: 11
+           :emphasize-lines: 6
 
            String path = "master-key.txt";
 
-           byte[] fileBytes = new byte[96];
+           byte[] localMasterKey= new byte[96];
 
            try (FileInputStream fis = new FileInputStream(path)) {
-             fileBytes = fis.readAllBytes();
-           } catch (IOException e) {
-             e.printStackTrace();
+               fis.readNBytes(localMasterKey, 0, 96);
```

Comment:
If you want to be really careful, you could check the return value and throw if 96 bytes were not read.

Reply:
Sure, that sounds like a good idea -- would you recommend throwing an Exception if it's <96 bytes?

I just discovered `readNBytes()` isn't in Java 1.8, so I will probably switch to `read()`
1.8: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html
1.9 https://docs.oracle.com/javase/9/docs/api/java/io/InputStream.html

Reply:
read is problematic because it can return before reading the requested number of bytes.  So you have to call it in a loop.  That's why these newer methods were added.  It's a PITA

Reply:
Since the 3.11 MongoDB driver is marked as compatible with Java 6, what would be your recommendation on reading the file into a byte array? Perhaps this snippet? https://stackoverflow.com/a/38235210

Reply:
There's just no good way to one-liner that does this before Java 9.  Not sure what to say

Reply:
Ok. I'll keep it as `readNBytes` and add an admonition that mentions that if they are using 8, they will need to implement that line differently, and link to stackoverflow or (hopefully) another resource.